### PR TITLE
Update projects to multi-target netstandard and net6

### PR DESCRIPTION
**Pull request by DrNeil**
Supersedes migrate projects to netstandard2.0 and net6.0 #1

### CODE REVIEW

**Comment by MichaelKetting** [comment]
@DrNeil 
I've triggered the workflow and there's some compatibility issues that you please need to resolve. From the looks of it, it seems to be mainly that you need to update the OS to windows-2022 to support the MSBuild 2022.

**Comment by MichaelKetting** [comment]
@DrNeil Since I don't know if you get the fail-notification: the automated build needs more work to make it compatible with .net 6 targets.

**Comment by glennawatson** [comment]
Hi @MichaelKetting I've taken over trying to get the build automation working for @DrNeil 

Just done some basic updates around attempting to install the latest .NET 6 etc

Let me know if you have any questions.

File: Rhino.Mocks/Rhino.Mocks.csproj [hunk 1]
```diff
@@ -13,9 +13,6 @@
     <NuspecProperties>version=$(Version)</NuspecProperties>
     <RepositoryUrl>https://github.com/rubicon-oss/CoreRhinoMocks</RepositoryUrl>
   </PropertyGroup>
-  <PropertyGroup>
-    <FrameworkPathOverride Condition="'$(TargetFramework)' == 'net35'">$(MSBuildProgramFiles32)\Reference Assemblies\Microsoft\Framework\.NETFramework\v3.5\Profile\Client</FrameworkPathOverride>
```
Comment (glennawatson) [inline Rhino.Mocks/Rhino.Mocks.csproj:17]: This seems to be the incorrect folder on Windows 2022 devices.

File: Rhino.Mocks.sln [hunk 2]
```diff
@@ -1,65 +1,37 @@
...
-		{C63839EC-BEF3-4DCD-BE23-B6DA21E8BE20}.Release|Win32.ActiveCfg = Release|Any CPU
 		{1E8FECC7-9E9B-4275-A938-D956F5E5D4F8}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{1E8FECC7-9E9B-4275-A938-D956F5E5D4F8}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{1E8FECC7-9E9B-4275-A938-D956F5E5D4F8}.Debug|Mixed Platforms.ActiveCfg = Debug|Any CPU
-		{1E8FECC7-9E9B-4275-A938-D956F5E5D4F8}.Debug|Mixed Platforms.Build.0 = Debug|Any CPU
-		{1E8FECC7-9E9B-4275-A938-D956F5E5D4F8}.Debug|Win32.ActiveCfg = Debug|Any CPU
 		{1E8FECC7-9E9B-4275-A938-D956F5E5D4F8}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{1E8FECC7-9E9B-4275-A938-D956F5E5D4F8}.Release|Any CPU.Build.0 = Release|Any CPU
-		{1E8FECC7-9E9B-4275-A938-D956F5E5D4F8}.Release|Mixed Platforms.ActiveCfg = Release|Any CPU
-		{1E8FECC7-9E9B-4275-A938-D956F5E5D4F8}.Release|Mixed Platforms.Build.0 = Release|Any CPU
```
Comment (glennawatson) [inline Rhino.Mocks.sln:41]: We weren't compiling Mixed platforms/win32

File: .github/workflows/ci_build.yml [hunk 3]
```diff
@@ -8,13 +8,24 @@ on:
...
   build:
-    runs-on: windows-2019
+    runs-on: windows-2022
 
     steps:
-      - uses: actions/checkout@v2
+      - name: Check Out
+        uses: actions/checkout@v3.0.2
+
+      - name: Install .NET 6
```
Comment (glennawatson) [inline .github/workflows/ci_build.yml:17]: Install the latest net6/netcoreapp3.1, also updated the github actions to the latest versions.

Recommend you maybe consider using dependabot to get latest github versions (along with .net components)

To reduce spam I generally have Dependabot only check for updates once a month.

Comment (MichaelKetting) [reply .github/workflows/ci_build.yml:17]: Thanks for the tip. From the looks of it, Dependabot is just a config file in the repo so if you'd like, I'd be happy to receive the correct config as a PR.

Comment (glennawatson) [reply .github/workflows/ci_build.yml:17]: Would you like it as part of this pr or separate?

**Comment by MichaelKetting** [comment]
Thanks @glennawatson. Looks good now!

File: .github/workflows/ci_build.yml (hunk 3 above)
Comment (MichaelKetting) [reply .github/workflows/ci_build.yml:17]: A separate ticket and PR please. 

And I've just noticed, the CI build does do nuget pack, so there's a problem with the .NET 3.5 target framework stemming from https://github.com/rubicon-oss/CoreRhinoMocks/pull/5#discussion_r905664101:
https://github.com/rubicon-oss/CoreRhinoMocks/runs/7060597905?check_suite_focus=true#step:10:35

Comment (glennawatson) [reply .github/workflows/ci_build.yml:17]: I'll check out the pack tomorrow

Comment (MichaelKetting) [reply .github/workflows/ci_build.yml:17]: meanwhile: I'm currently updating the workflow so we can ci test the build process. should be integrated in tomorrow's develop.

**Pull request merged**